WordPress/src/test/java/org/wordpress/android/viewmodel/helpers/ConnectionStatusLiveDataTest.kt

```diff
@@ -0,0 +1,95 @@
+package org.wordpress.android.viewmodel.helpers
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.arch.lifecycle.LiveData
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.mock
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.wordpress.android.viewmodel.helpers.ConnectionStatus.AVAILABLE
+import org.wordpress.android.viewmodel.helpers.ConnectionStatus.UNAVAILABLE
+
+class ConnectionStatusLiveDataTest {
```

Comment:
I'm wondering whether we want to add `@RunWith(MockitoJUnitRunner::class)` runner? We don't use `@Mock` nor `whenever` or similar so I'm not sure if it has any advantages. We might also consider extending `BaseUnitTest`. Wdyt?

Reply:
> I'm wondering whether we want to add @RunWith(MockitoJUnitRunner::class) runner? We don't use @Mock nor whenever or similar so I'm not sure if it has any advantages.

That's interesting. I somehow survived without ever using this class. I'll try it out next time. I'm not sure if this test class will survive if we go with the factory route. 

> We might also consider extending BaseUnitTest

I'm not sure. If it's required by the team, then okay. I'm somewhat against using base classes for tests because they make the test dependencies and execution unclear. The `InstantTaskExecutorRule` addition is just one line after all. And there's also that possibility of using [RuleChain](https://github.com/junit-team/junit4/wiki/Rules#rulechain) which may not be possible with a base class.

Reply:
>I'm not sure. If it's required by the team, then okay. I'm somewhat against using base classes for tests because they make the test dependencies and execution unclear. The InstantTaskExecutorRule addition is just one line after all. And there's also that possibility of using RuleChain which may not be possible with a base class.

💯 

>That's interesting. I somehow survived without ever using this class. I'll try it out next time. I'm not sure if this test class will survive if we go with the factory route.

I think it's not required but it gives you additional warnings if you for example mock an object but you never use it in the test. It can be also used to inject/init mocks annotated with @Mock annotation. As I mentioned it might be useless in this test class, I just wanted to mention it as I wasn't sure whether it wasn't missed unintentionally. ;)